04.class/lib/memo_app.js

```diff
@@ -0,0 +1,86 @@
+import { Memo } from "./memo.js";
+import { View } from "./view.js";
+
+export class MemoApp {
+  #view = null;
+
+  constructor() {
+    this.#view = new View();
+  }
+
+  async run(params) {
+    await Memo.createTable();
+
+    if (params.l) {
+      await this.#printMemos();
+    } else if (params.r) {
+      await this.#printMemo();
+    } else if (params.d) {
+      await this.#deleteMemo();
+    } else {
+      await this.#createMemo();
+    }
+
+    await Memo.close();
+  }
+
+  async #printMemos() {
+    const memos = await Memo.all();
+    if (memos.length === 0) {
+      this.#view.printNoMemo();
+    } else {
+      this.#view.printMemos(memos);
+    }
+  }
+
+  async #printMemo() {
+    const memos = await Memo.all();
+    if (memos.length === 0) {
+      this.#view.printNoMemo();
+      return;
+    }
+
+    const selectedTitle = await this.#view.displayMemoSelectionPrompt(
+      memos,
+      "r",
+    );
+
+    const selectedMemo = await Memo.findByTitle(selectedTitle);
```

Comment:
```suggestion
    const memo = await Memo.findByTitle(selectedTitle);
```

これだけで十分かも。
https://bootcamp.fjord.jp/pages/avoid-too-specific-var-name